docs(self-hosted): enabling preview features section (#1148)
getsentry/develop: src/docs/self-hosted/index.mdx

  Once you change your configuration, you'll need to restart all Sentry services by running <code>docker compose restart web worker cron sentry-cleanup</code> (or just <code>docker compose restart</code> to restart everything).
</Alert>

### Enabling Preview Features

> Only features that has reached General Availability (GA) status will be enabled by default on self-hosted. As users, you can try new features before it reaches GA by adding required component and configurations. Preview features are tracked and documented through [GitHub issues with Type: Pre-release Feature labels](https://github.com/getsentry/self-hosted/labels/Type%3A%20Pre-release%20Feature).

Reviewer: ```suggestion
Only features that have reached General Availability (GA) status will be enabled by default for self-hosted. As users, you can try new features before it reaches GA by adding required infrastructure components and feature flags. Preview features are tracked and documented through [GitHub issues with Type: Pre-release Feature labels](https://github.com/getsentry/self-hosted/labels/Type%3A%20Pre-release%20Feature).
```

Reviewer: Otherwise, looks good to me :)
